osu.Game.Tests/Visual/SongSelect/TestScenePlaySongSelect.cs

```diff
@@ -73,6 +74,17 @@ public override void SetUpSteps()
             AddStep("delete all beatmaps", () => manager?.Delete());
         }
 
+        public override void TearDownSteps()
+        {
+            base.TearDownSteps();
+
+            AddStep("remove song select", () =>
+            {
+                songSelect?.Expire();
+                songSelect = null;
+            });
```

Comment:
Why is this required/why doesn't `base.TearDownSteps()` forcefully exiting screens in the stack handle this?

Reply:
Ah, didn't realise this was a screen test.

The important part is the `null` set, so the next run doesn't see the previous run's (expired) `songSelect` and try to match against it (see the special case of the newly adding condition, where we aren't always guaranteed the song select has been created yet).

Can reduce this just to the null set if that reads better.

Reply:
Maybe best to do that null set in `Setup` or `SetupSteps`?

Reply:
Sure, will move it. Had that in a previous iteration.